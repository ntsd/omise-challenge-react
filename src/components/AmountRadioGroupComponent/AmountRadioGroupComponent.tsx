import React from 'react';
import {Form} from 'react-bootstrap';

interface AmountRadioGroupComponentProps {
  selectedAmount: number;
  setSelectedAmount: React.Dispatch<React.SetStateAction<number>>;
}

export const AmountRadioGroupComponent: React.FC<AmountRadioGroupComponentProps> = ({
  selectedAmount,
  setSelectedAmount,
}) => {
  const donateList = [10, 20, 50, 100, 500];

  return (
    <div>
      {donateList.map((amount, index) => (
        <Form.Check
          inline
          type="radio"
          label={amount}
          value={amount}
          checked={selectedAmount === amount}
          key={'radio-' + index}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
            setSelectedAmount(parseInt(event.target.value));
          }}
        />
      ))}
    </div>
  );
};
