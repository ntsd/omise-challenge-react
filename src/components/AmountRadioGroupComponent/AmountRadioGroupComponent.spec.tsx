import React, {useState} from 'react';
import renderer from 'react-test-renderer';
import {AmountRadioGroupComponent} from './AmountRadioGroupComponent';

it('renders correctly', () => {
  const TestElement = () => {
    const [selectedAmount, setSelectedAmount] = useState(10);
    return (
      <AmountRadioGroupComponent
        selectedAmount={selectedAmount}
        setSelectedAmount={setSelectedAmount}
      ></AmountRadioGroupComponent>
    );
  };

  const tree = renderer.create(<TestElement />).toJSON();
  expect(tree).toMatchSnapshot();
});
