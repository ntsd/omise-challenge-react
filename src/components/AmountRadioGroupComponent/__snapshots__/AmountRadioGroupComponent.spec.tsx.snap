// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="form-check form-check-inline"
    style={undefined}
  >
    <input
      checked={true}
      className="form-check-input"
      disabled={false}
      id={undefined}
      onChange={[Function]}
      type="radio"
      value={10}
    />
    <label
      className="form-check-label"
      htmlFor={undefined}
      title=""
    >
      10
    </label>
  </div>
  <div
    className="form-check form-check-inline"
    style={undefined}
  >
    <input
      checked={false}
      className="form-check-input"
      disabled={false}
      id={undefined}
      onChange={[Function]}
      type="radio"
      value={20}
    />
    <label
      className="form-check-label"
      htmlFor={undefined}
      title=""
    >
      20
    </label>
  </div>
  <div
    className="form-check form-check-inline"
    style={undefined}
  >
    <input
      checked={false}
      className="form-check-input"
      disabled={false}
      id={undefined}
      onChange={[Function]}
      type="radio"
      value={50}
    />
    <label
      className="form-check-label"
      htmlFor={undefined}
      title=""
    >
      50
    </label>
  </div>
  <div
    className="form-check form-check-inline"
    style={undefined}
  >
    <input
      checked={false}
      className="form-check-input"
      disabled={false}
      id={undefined}
      onChange={[Function]}
      type="radio"
      value={100}
    />
    <label
      className="form-check-label"
      htmlFor={undefined}
      title=""
    >
      100
    </label>
  </div>
  <div
    className="form-check form-check-inline"
    style={undefined}
  >
    <input
      checked={false}
      className="form-check-input"
      disabled={false}
      id={undefined}
      onChange={[Function]}
      type="radio"
      value={500}
    />
    <label
      className="form-check-label"
      htmlFor={undefined}
      title=""
    >
      500
    </label>
  </div>
</div>
`;
